web/pgadmin/tools/sqleditor/static/js/components/sections/MainToolBar.jsx

```diff
@@ -89,14 +90,21 @@ export function MainToolBar({containerRef, onFilterClick, onManageMacros}) {
   }, []);
 
   const executeQuery = useCallback(()=>{
+    if(!queryToolCtx.preferences.sqleditor.underline_query_cursor && queryToolCtx.preferences.sqleditor.underline_query_cursor_warning){
```

Comment:
the useCallback will not use latest queryToolCtx.preferences. You should add dependency.